src/hooks/use-fetch-wpcom-sites.tsx

```diff
@@ -178,15 +302,14 @@ export const useFetchWpComSites = ( connectedSiteIds: number[] ) => {
 		return fetchSites();
 	}, [ fetchSites ] );
 
-	const syncSites = useMemo(
+	const syncSitesWithSyncSupportForSelectedSite = useMemo(
 		() => transformSiteResponse( rawSyncSites, memoizedConnectedSiteIds ),
 		[ rawSyncSites, memoizedConnectedSiteIds ]
 	);
```

Comment:
Ideally we should get rid of these lines later in the refactoring PR.
Also, we should get rid of `connectedSiteIdsOnlyForSelectedSite` argument.

It's not easy to explain this in writing, since the code around it is very fragile and bound.
In a nutshell, we are rendering the list of sites in connect-dialog. Every item has own status (syncable, unsupported, etc). And the main problem - if some site is already connected to the selected site - we retrieve this information from `useFetchWpComSites` hood, by providing the list of `connectedSiteIdsOnlyForSelectedSite`. So, we [modify](https://github.com/Automattic/studio/blob/nightnei/fixOcasionallyMarkedSiteAsDeleted/src/hooks/use-fetch-wpcom-sites.tsx#L207) the list of `syncSites` everytime, based on selected domain. It means that our list of sites is not stable and as a result it was the problem with occasionally market site as deleted, since during switching between sites in Studio UI we updated syncSupport in appdata-v1.json.

I temporary fixed it with two steps:
1) [Update](https://github.com/Automattic/studio/blob/nightnei/fixOcasionallyMarkedSiteAsDeleted/src/hooks/use-fetch-wpcom-sites.tsx#L256-L283) appdata-v1.json as the first step only when we fetch sites
2) Every time when we run hook with different list of provided connected ids - I modify returned array.
As a result - it works totally as previously, but updating of appdata-v1.json moved to 1 level down and is not connected to the code outside of it.

It's difficult to understand, so I will be happy to have a quick huddle to show the issue with occasionally deleted sites.

Reply:
I think [my comment here](https://github.com/Automattic/studio/pull/749#discussion_r1890388013) essentially addresses the same thing.

Reply:
Yes, exactly, it's about the same, but I decided to keep PRs more specific, so in this PR I want to fix issue which we already have in UI, but this refactoring wanted to do separately (mentioned it p1734458315877079-slack-C04GESRBWKW), as a tech-debt.